### PR TITLE
Add istio-workspace 0.1 tasks

### PR DESCRIPTION
**Pull request by alien-ike**
# Changes

* Add Task ike-create release 0.1
* Add Task ike-delete release 0.1
* Add Task ike-session-url release 0.1

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/ike-create/0.1/ike-create.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: ike-create
+  labels:
+    app.kubernetes.io/version: 0.1
```
Comment (alien-ike) [inline task/ike-create/0.1/ike-create.yaml:6]:
```suggestion
    app.kubernetes.io/version: "0.1"
```

File: task/ike-delete/0.1/ike-delete.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: ike-delete
+  labels:
+    app.kubernetes.io/version: 0.1
```
Comment (alien-ike) [inline task/ike-delete/0.1/ike-delete.yaml:6]:
```suggestion
    app.kubernetes.io/version: "0.1"
```

File: task/ike-session-url/0.1/ike-session-url.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: ike-session-url
+  labels:
+    app.kubernetes.io/version: 0.1
```
Comment (alien-ike) [inline task/ike-session-url/0.1/ike-session-url.yaml:6]:
```suggestion
    app.kubernetes.io/version: "0.1"
```

Comment (alien-ike) [reply task/ike-session-url/0.1/ike-session-url.yaml:6]: @chmouel I guess that's another reason why semver would be less painful to deal with ;)

**Comment by chmouel** [comment]
/lgtm

I have the feeling that the README looks a bit light, 

you may want to address some of the linter shellcheck errors (not sure why the job is not working) 

```
% catlin lint script ike-*/*/*.yaml
ERROR: /usr/bin/shellcheck, [-s sh] failed:

In ike-create-ike line 2:
test ! -z "$(params.route)" && export ROUTE="--route $(params.route)"
                                      ^---^ SC2155: Declare and assign separately to avoid masking return values.


In ike-create-ike line 4:
STATE=`ike create --session $(params.session) $ROUTE --deployment $(params.target) --namespace $(params.namespace) --image $(params.image) --json`
      ^-- SC2006: Use $(...) notation instead of legacy backticked `...`.

Did you mean: 
STATE=$(ike create --session $(params.session) $ROUTE --deployment $(params.target) --namespace $(params.namespace) --image $(params.image) --json)


In ike-create-ike line 5:
if [[ $? -ne 0 ]]; then
   ^------------^ SC2039: In POSIX sh, [[ ]] is undefined.
      ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

For more information:
  https://www.shellcheck.net/wiki/SC2039 -- In POSIX sh, [[ ]] is undefined.
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
  https://www.shellcheck.net/wiki/SC2006 -- Use $(...) notation instead of le...

ERROR: /usr/bin/shellcheck, [-s sh] failed:

In ike-session-url-oc line 2:
STATE=`oc get session $(params.session) --namespace $(params.namespace) -o json`
      ^-- SC2006: Use $(...) notation instead of legacy backticked `...`.

Did you mean: 
STATE=$(oc get session $(params.session) --namespace $(params.namespace) -o json)


In ike-session-url-oc line 3:
if [[ $? -ne 0 ]]; then
   ^------------^ SC2039: In POSIX sh, [[ ]] is undefined.
      ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

For more information:
  https://www.shellcheck.net/wiki/SC2039 -- In POSIX sh, [[ ]] is undefined.
  https://www.shellcheck.net/wiki/SC2006 -- Use $(...) notation instead of le...
  https://www.shellcheck.net/wiki/SC2181 -- Check exit code directly with e.g...

```

File: task/ike-create/0.1/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+-   `session`(string): What to call the session.
+
+-   `route`(string): How to route.
+
+    -   default value:
+
+-   `image`(string): The image to use for the route.
+
+### Results
```
Comment (chmouel) [inline task/ike-create/0.1/README.md:23]: @vinamra28 did we decide on Results+workspace or either ?

**Comment by alien-ike** [comment]
> you may want to address some of the linter shellcheck errors (not sure why the job is not working)

Sure, let me work on that in this PR.

File: task/ike-create/0.1/README.md (hunk 4 above)
Comment (vinamra28) [reply task/ike-create/0.1/README.md:23]: I guess we can continue with both `Results+workspace` for now

File: task/ike-create/0.1/ike-create.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      script: |
+        test ! -z "$(params.route)" && export ROUTE="--route $(params.route)"
+
+        STATE=`ike create --session $(params.session) $ROUTE --deployment $(params.target) --namespace $(params.namespace) --image $(params.image) --json`
+        if [[ $? -ne 0 ]]; then
+          error=$?
+          echo $STATE
+          exit $error
+        fi
+        echo "$STATE" | grep \"hosts\": | cut -d \" -f 4 | tr -d '\n' > /tekton/results/url
```
Comment (alien-ike) [inline task/ike-create/0.1/ike-create.yaml:48]:
```suggestion
        test ! -z "$(params.route)" && ROUTE="--route $(params.route)"
        export ROUTE

        if ! STATE=$(ike create --session $(params.session) $ROUTE --deployment $(params.target) --namespace $(params.namespace) --image $(params.image) --json); then
          error=$?
          echo $STATE
          exit $error
        fi
        echo "$STATE" | grep \"hosts\": | cut -d \" -f 4 | tr -d '\n' > /tekton/results/url

```

File: task/ike-session-url/0.1/ike-session-url.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    - name: oc
+      image: quay.io/openshift/origin-cli:4.9
+      script: |
+        STATE=`oc get session $(params.session) --namespace $(params.namespace) -o json`
+        if [[ $? -ne 0 ]]; then
+          error=$?
+          echo $STATE
+          exit $error
+        fi
+        echo "$STATE" | grep \"hosts\": | cut -d \" -f 4 | uniq | tr -d '\n' > /tekton/results/url
```
Comment (alien-ike) [inline task/ike-session-url/0.1/ike-session-url.yaml:36]:
```suggestion
        if ! STATE=$(oc get session $(params.session) --namespace $(params.namespace) -o json); then
          error=$?
          echo $STATE
          exit $error
        fi
        echo "$STATE" | grep \"hosts\": | cut -d \" -f 4 | uniq | tr -d '\n' > /tekton/results/url

```

**Comment by alien-ike** [comment]
Closing in favor of #726

**Pull request closed without merging**